mysql-test/suite/rpl/t/rpl_show_slave_hosts.test

```diff
@@ -49,4 +49,497 @@ source include/wait_show_condition.inc;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+--sync_with_master
+
+connection slave2;
+--source include/start_slave.inc
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+--sync_with_master
+
+--echo # sync with master, should ENABLE master_status, commit acknowledgment will be cleared?
+connection master;
+let $status_var= Rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_status';
+show status like 'Rpl_semi_sync_master_no_tx';
+show status like 'Rpl_semi_sync_master_yes_tx';
+show status like 'rpl_semi_sync_master_get_ack';
+
+--echo # Test 10: Testing `rpl_semi_sync_master_timeout=0`
+--echo # Note 1:  For asynchronous replication `Gtid_state_sent` is not null,
+--echo #          but `Gtid_state_ack` is null.
+--echo # Note 2:  We want that `rpl_semi_sync_master_timeout=0` be edge case,
+--echo #          by mimicking asynchronous replication, as semi-sync stalled
+--echo #          replication type, for which `Gtid_state_ack` should not update
+--echo #          or be null (see Note 3). Also `rpl_semi_sync_master_status` has
+--echo #          to remain `ON`, instead switching off, so instead of
+--echo #          no waiting on ACK, primary must wait on ACK from replica.
+--echo # Note 3:  With this option we want `Gtid_state_ack` to reflect
+--echo #          the latest received transaction by the replica[s].
+--echo # -----------------------------------------------------------------------
+# Test DBUG_SYNC by stopping the slave and testing the new sync point
+# Note we need to stop both slaves in order to control sent ACK for new active transaction
+
+--let $save_default_master_timeout= `SELECT @@GLOBAL.rpl_semi_sync_master_timeout`
+select @@global.rpl_semi_sync_master_timeout;
+show variables like 'rpl_semi_sync_master_enabled';
+
+# Master status must be `ON`
+let $status_var= rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_master_status';
+
+show status like 'rpl_semi_sync_master_get_ack';
+set global rpl_semi_sync_master_timeout=0;
+select @@global.rpl_semi_sync_master_timeout;
+# here rpl_semi_sync_master_status=0 means async is taking place (no wait on ACK from slave)
+show variables like 'rpl_semi_sync_master_timeout';
+show status like 'rpl_semi_sync_master_status';
+
+connection slave;
+--source include/stop_slave.inc
+--let $save_server_2_dbug= `SELECT @@GLOBAL.DEBUG_DBUG`
+SET @@GLOBAL.DEBUG_DBUG="d,synchronize_semisync_slave_reply";
+--source include/start_slave.inc
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'rpl_semi_sync_slave%';
+SET debug_sync='now WAIT_FOR at_slave_reply TIMEOUT 10';
```

Comment:
Here (and slave2's line 458) time out (see lines 427 and 428 in the result file), as there have been no transactions yet to acknowledge from the slave, so the debug sync points should not be called. These synchronizations points should go right after line 463, as at that point the replicas will need to acknowledge the new event. 

Looking at my [previous comment](https://github.com/MariaDB/server/pull/2374/files#r1213731775) it looks I commented at the wrong spot to put it, sorry.